### PR TITLE
E1716. Improve e-mail notifications

### PR DESCRIPTION
**Pull request by manoj4269**

### CODE REVIEW

**Comment by Winbobob** [comment]
Hi team,

Please revert the changes on `rubocop.yml` and `.travisci.yml`.

Thanks,
Zhewei

File: app/controllers/invitation_controller.rb [hunk 1]
```diff
@@ -9,8 +9,11 @@ def new
 
   def create
     user = User.find_by_name(params[:user][:name].strip)
+
     team = AssignmentTeam.find(params[:team_id])
     student = AssignmentParticipant.find(params[:student_id])
+    student_email = student.user_id
```
Comment (efg) [inline app/controllers/invitation_controller.rb]: What?  User_ID normally doesn't contain the domain, but email needs to.  Sounds like a bad variable name, or worse.

File: app/controllers/invitation_controller.rb [hunk 2]
```diff
@@ -37,6 +40,7 @@ def create
             @invitation.assignment_id = student.parent_id
             @invitation.reply_status = 'W'
             @invitation.save
+            accept_invitation(student_email)  # Call the mailing function
```
Comment (efg) [inline app/controllers/invitation_controller.rb]: The mailing function is called accept_invitation?  This is really confusing.

File: app/controllers/invitation_controller.rb [hunk 3]
```diff
@@ -65,8 +79,20 @@ def auto_complete_for_user_name
     @users = User.where("LOWER(name) LIKE ?", "%#{search}%") unless search.blank?
   end
 
+  def accepted_invitation(user_email)
```
Comment (efg) [inline app/controllers/invitation_controller.rb]: Another badly named method.  Should be email_an_acceptance

Comment (manoj4269) [inline app/controllers/invitation_controller.rb]: Hello Prof, we have fixed all the naming conventions that you have suggested us.
accept_invitation => email_an_invitation
accepted_invitation => email_an_acceptance
student_email => student_id

**Comment by manoj4269** [comment]
Hello Zhewei
We have refactored and cleared out the issues in the code climate. But some of the issues showing up are from the code written earlier in some other controllers which we haven't touched at all and some are from the mailer which was also present earlier and can't be changed. 
Thanks, 
Manoj

File: Gemfile [hunk 4]
```diff
@@ -66,7 +67,6 @@ group :development do
   gem 'pry'
   gem 'pry-remote'
   gem 'pry-nav'
-  gem 'rubocop'
```
Comment (Winbobob) [inline Gemfile]: Why do you delete this gem?

File: app/helpers/import_file_helper.rb [hunk 5]
```diff
@@ -18,6 +18,9 @@ def self.create_new_user(attributes, session)
     user.parent_id = (session[:user]).id
     user.timezonepref = User.find(user.parent_id).timezonepref
     user.save!
+    password = 'qwerty' # the password is reset
```
Comment (Winbobob) [inline app/helpers/import_file_helper.rb]: I remember the password is auto-generated instead of hard coded after importing new students.

**Pull request closed without merging**